OptLib/Functions/Function.h

```diff
@@ -143,14 +143,11 @@ namespace OptLib
 			}
 		};
 
-
-
-
 		template <size_t dim>
 		class FuncAlongGradDirection : public FuncInterface::IFuncWithGrad<1>
 		{
 		public:
-			FuncAlongGradDirection(FuncInterface::IFuncWithGrad<dim>* f_pointer, const Point<dim>& x0_) noexcept :
+			FuncAlongGradDirection(const FuncInterface::IFuncWithGrad<dim>* f_pointer,Point<dim>& x0_) noexcept :
```

Comment:
What the reason to pass const pointer in this case? It will be converted to link so why not just pass link?

Reply:
The reason to pass const pointer in this case is parameters of the Proceed method in each optimization method.  All of them contain const pointer. Where will it be converted to a link?

Reply:
The reason to pass const pointer in this case is parameters of the Proceed method. All of them contain const pointer. Where will it be be converted to link?

Reply:
row 169 const FuncInterface::IFuncWithGrad<dim>&  f;, row 151 f{ *f_pointer }. Its kinda useless conversion since you can just pass link or store prt instead

Reply:
The reason to pass const pointer in this case is parameters of the Proceed methods. All of them contain const pointer. Where will it be converted to link?

Reply:
Yes, you're right. I will send a commit